articles/ros2_dds_security.md

```diff
@@ -135,52 +135,66 @@ For example, for the `/front/camera` node, the directory structure would look li
-- **permissions.p7s**: The XML document that specifies the permissions of this particular node instance to the **Access control** plugin (also signed by the Permissions CA).
+- **permissions.p7s**: The XML document that specifies the permissions of this particular context instance to the **Access control** plugin (also signed by the Permissions CA).
+
+This can be specified by setting the `$ROS_SECURITY_ROOT_DIRECTORY` environment variable to point to the root of the contexts directory tree, and then specifying the context path using the `--ros-args` runtime argument `--contex`, e.g.:
 
-This can be specified by setting the `$ROS_SECURITY_ROOT_DIRECTORY` environment variable to point to the root of the directory tree.
+``` shell
+export ROS_SECURITY_ROOT_DIRECTORY="/home/bob/.ros/sros2_keystore/contexts"
+ros2 run <package> <executable> --ros-args --contex="/front/camera"
+```
 
 
 ##### Support security files lookup methods
 
-If using the directory tree approach to organize security files, RCL supports two different methods for looking up a given node instance's security files in the tree:
+If using the directory tree approach to organize security files, RCL supports two different methods for looking up a given context instance's security files in the tree:
 
-- **Exact**: Only load security files from a directory exactly matching the fully-qualified name of the node instance.
-For example, given a node named "baz_123" within the "/foo/bar/" namespace, only load security files from `<root>/foo/bar/baz_123/`.
+- **Exact**: Only load security files from a directory exactly matching the fully-qualified path of the context.
+For example, given a context "baz_123" within the "/foo/bar/" subpath, only load security files from `<root>/foo/bar/baz_123/`.
 This is the default behavior.
-- **Prefix**: Attempt to load the most specific set of security files, but if they can't be found, check for security files under a less-specific node name.
-For example, given a node named "baz_123" within the "/foo/bar/" namespace, load security files from `<root>/foo/bar/baz_123/`.
-However, if that directory doesn't exist, find the most specific (i.e. longest) node name that _does_ have security files within that namespace (e.g. `<root>/foo/bar/baz_12/`, or `<root>/foo/bar/baz/`, etc.).
-Note that it will not search higher in the namespace hierarchy.
+- **Prefix**: Attempt to load the most specific set of security files, but if they can't be found, check for security files under a less-specific context path.
+For example, given a context "baz_123" within the "/foo/bar/" subpath, load security files from `<root>/foo/bar/baz_123/`.
+However, if that directory doesn't exist, find the most specific (i.e. longest) context path that _does_ have security files within that subpath (e.g. `<root>/foo/bar/baz_12/`, or `<root>/foo/bar/baz/`, etc.).
+Note that it will not search higher in the subpath hierarchy.
```

Comment:
Unrelated question, out of curiosity: which use cases have the `Prefix` look up method?

Reply:
I'd guess it'd still be used when the user provides a context path to `--context` arg and would like RCL to gracefully fallback to using the context path with the longest matching prefix. E.g.

```
$ mkdir -p ${ROS_SECURITY_ROOT_DIRECTORY}/foo/bar
$ cp <security artifacts> !$
$ export ROS_SECURITY_LOOKUP_TYPE="Prefix"
$ ros2 run <package> <executable> --ros-args --context="/foo/bar/baz"
# loads security artifacts from context path /foo/bar
```

I think having alternate look up strategies is a bit over complicated, and is another pattern that obscures the relation between policy definition and runtime configuration, but at least it isn't default. You can find more discussion on its introduction here: https://github.com/ros2/rcl/pull/332

Reply:
Yeah in my mind the new design allowed us to drop the lookup type + tinydir dependency completely.

In the past there was a need for it as some nodes could not be defined in a policy fiule without it (ros2 cli or any nodes using patterns like `ros2cli_UNPREDICTIBLE_STUFF` for nodes created on the fly at runtime)

As now every case can be defined in a policy file it seems more like a deceiving fallback than a convenient/solving a problem fallback.

Reply:
> As now every case can be defined in a policy file it seems more like a deceiving fallback than a convenient/solving a problem fallback.

Agreed, now that artifacts aren't bound to dynamic node names, `Prefix` seems redundant.

Reply:
from WG meeting: let's ditch it and the tinydir dependency. We can add it back if a compelling use case for it is brought up

Reply:
Use of alternate lookup methods have been remove in 2a2122b

Reply:
Use of alternate lookup methods have been remove in 2a2122b